src/test/java/code401challenges/QuickSortTest.java

```diff
@@ -11,7 +13,8 @@
     public void quickSortTest() {
         int[] test = {4,5,1,6,7,4,2,9,3,6};
         int[] ans = {1,2,3,4,4,5,6,6,7,9};
-        assertArrayEquals(ans, quickSort(test, test[0], test[test.length-1]));
+        System.out.println(Arrays.toString(quickSort(test, 0, test.length-1)));

```

Comment:
This is a good test... but you have to test some edge cases to verify your method is working the way you want. Edge cases like: length 0 array, length 1 array, a pre-sorted array, and reverse sorted array.